modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+			}
+			$new_filter                                   = $filter;
+			$new_filter['filter_id']                      = $key;
+			$widgets[ $filter['widget_id'] ]['filters'][] = $new_filter;
+		}
+
+		$post_type_objs   = get_post_types( array(), 'objects' );
+		$post_type_labels = array();
+		foreach ( $post_type_objs as $key => $obj ) {
+			$post_type_labels[ $key ] = array(
+				'singular_name' => $obj->labels->singular_name,
+				'name'          => $obj->labels->name,
+			);
+		}
+
+		$prefix  = Jetpack_Search_Options::OPTION_PREFIX;
+		$options = array(
+			'overlayOptions'  => array(
+				'closeColor'      => get_option( $prefix . 'close_color', '#BD3854' ),
+				'colorTheme'      => get_option( $prefix . 'color_theme', 'light' ),
+				'enableInfScroll' => (bool) get_option( $prefix . 'inf_scroll', false ),
+				'highlightColor'  => get_option( $prefix . 'highlight_color', '#FFC' ),
+				'opacity'         => (int) get_option( $prefix . 'opacity', 97 ),
+				'showLogo'        => (bool) get_option( $prefix . 'show_logo', true ),
+				'showPoweredBy'   => (bool) get_option( $prefix . 'show_powered_by', true ),
+			),
+
+			// core config.
+			'homeUrl'         => home_url(),
+			'locale'          => str_replace( '_', '-', get_locale() ),
+			'postsPerPage'    => get_option( 'posts_per_page' ),
+			'siteId'          => Jetpack::get_option( 'id' ),
+
+			// filtering.
+			'postTypeFilters' => $widget_options['post_types'],
+			'postTypes'       => $post_type_labels,
+			'sort'            => $widget_options['sort'],
+			'widgets'         => array_values( $widgets ),
+		);
+
+		/**
+		 * Customize Instant Search Options.
+		 *
+		 * @module search
+		 *
+		 * @since 7.7.0
+		 *
+		 * @param array $options Array of parameters used in Instant Search queries.
+		 */
+		$options = apply_filters( 'jetpack_instant_search_options', $options );
+
+		// Use wp_add_inline_script instead of wp_localize_script, see https://core.trac.wordpress.org/ticket/25280.
+		wp_add_inline_script( 'jetpack-instant-search', 'var JetpackInstantSearchOptions=' . wp_json_encode( $options ) );
+	}
+
+	/**
+	 * Registers a widget sidebar for Instant Search.
+	 */
+	public function register_jetpack_instant_sidebar() {
+		$args = array(
+			'name'          => 'Jetpack Search Sidebar',
+			'id'            => 'jetpack-instant-search-sidebar',
+			'description'   => 'Customize the sidebar inside the Jetpack Search overlay',
+			'class'         => '',
+			'before_widget' => '<div id="%1$s" class="widget %2$s">',
+			'after_widget'  => '</div>',
+			'before_title'  => '<h2 class="widgettitle">',
+			'after_title'   => '</h2>',
+		);
+		register_sidebar( $args );
+	}
+
+	/**
+	 * Prints Instant Search sidebar.
+	 */
+	public function print_instant_search_sidebar() {
+		?>
+		<div class="jetpack-instant-search__widget-area" style="display: none">
+			<?php if ( is_active_sidebar( 'jetpack-instant-search-sidebar' ) ) { ?>
```

Comment:
should we move the if check before the div? Do we want the div still in place if the `is_active_sidebar` check fails?

Reply:
It doesn't matter either way for the Instant Search JS client. Do you think we should change it?

Reply:
I'd suggest removal all together as it can also reduce js overhead if any actions based on the class. But this is fully at your discretion